spec/models/metric_spec.rb

```diff
@@ -1107,6 +1107,7 @@
         @vms_in_az = []
         2.times { @vms_in_az << FactoryGirl.create(:vm_openstack, :ems_id => @ems_openstack.id) }
         @availability_zone.vms = @vms_in_az
+        @availability_zone.vms.push(FactoryGirl.create(:vm_openstack, :ems_id => nil))
```

Comment:
@jrafanie I believe so.

What I'm reading is that there are now 3 OpenStack VMs ([2 here](https://github.com/ManageIQ/manageiq/pull/8429/files#diff-7d1b2d2b7780f53859ecf3e13d28a903R1108), and [1 here](https://github.com/ManageIQ/manageiq/pull/8429/files#diff-7d1b2d2b7780f53859ecf3e13d28a903R1110)) in the availability zone and 3 OpenStack VMs ([3 here](https://github.com/ManageIQ/manageiq/pull/8429/files#diff-7d1b2d2b7780f53859ecf3e13d28a903R1113)) outside the availability zone.

But, the assert only wants to find 5 OpenStack Vms [here](https://github.com/ManageIQ/manageiq/pull/8429/files#diff-7d1b2d2b7780f53859ecf3e13d28a903R1121).

So, even though an extra VM was added to the test, it falls outside the target selection criteria, so it's not matched.